libcontainer/integration/exec_test.go

```diff
@@ -1660,6 +1661,70 @@ func TestPIDHost(t *testing.T) {
 	}
 }
 
+func TestPIDHostInitProcessWait(t *testing.T) {
+	if testing.Short() {
+		return
+	}
+
+	rootfs, err := newRootfs()
+	ok(t, err)
+	defer remove(rootfs)
+
+	pidns := "/proc/1/ns/pid"
+
+	config := newTemplateConfig(rootfs)
+	config.Namespaces.Add(configs.NEWPID, pidns)
+	container, err := newContainerWithName("test", config)
+	ok(t, err)
+	defer func() {
+		err := container.Destroy()
+		ok(t, err)
+	}()
+
+	// Run a container with two long-running processes.
+	stdinR1, stdinW1, err := os.Pipe()
+	ok(t, err)
+	process1 := &libcontainer.Process{
+		Cwd:   "/",
+		Args:  []string{"cat"},
+		Env:   standardEnvironment,
+		Stdin: stdinR1,
+		Init:  true,
+	}
+	err = container.Run(process1)
+	ok(t, err)
+	stdinR1.Close()
+	defer stdinW1.Close()
+
+	stdinR2, stdinW2, err := os.Pipe()
+	ok(t, err)
+	process2 := &libcontainer.Process{
+		Cwd:   "/",
+		Args:  []string{"cat"},
+		Env:   standardEnvironment,
+		Stdin: stdinR2,
+		Init:  false,
+	}
+	err = container.Run(process2)
+	ok(t, err)
+	stdinR2.Close()
+	defer stdinW2.Close()
+
+	// Kill the init process and Wait for it.
+	err = process1.Signal(syscall.SIGKILL)
+	ok(t, err)
+	_, err = process1.Wait()
+	if err == nil {
+		t.Fatal("expected Wait to indicate failure")
+	}
+
+	// The non-init process must've been killed.
+	err = process2.Signal(syscall.Signal(0))
+	if err == nil || err.Error() != "no such process" {
+		t.Fatalf("expected process to have been killed: %q", err)
```

Comment:
```suggestion
		t.Fatalf("expected process to have been killed: %v", err)
```
As otherwise when it fails it prints
```
    exec_test.go:1721: expected process to have been killed: %!q(<nil>)
```